sway/config/output.c

```diff
@@ -153,28 +153,22 @@ static void merge_wildcard_on_all(struct output_config *wildcard) {
 }
 
 static void merge_id_on_name(struct output_config *oc) {
-	char *id_on_name = NULL;
-	char id[128];
-	char *name = NULL;
-	struct sway_output *output;
-	wl_list_for_each(output, &root->all_outputs, link) {
-		name = output->wlr_output->name;
-		output_get_identifier(id, sizeof(id), output);
-		if (strcmp(name, oc->name) == 0 || strcmp(id, oc->name) == 0) {
-			size_t length = snprintf(NULL, 0, "%s on %s", id, name) + 1;
-			id_on_name = malloc(length);
-			if (!id_on_name) {
-				sway_log(SWAY_ERROR, "Failed to allocate id on name string");
-				return;
-			}
-			snprintf(id_on_name, length, "%s on %s", id, name);
-			break;
-		}
+	struct sway_output *output = all_output_by_name_or_id(oc->name);
+	if (output == NULL) {
+		return;
 	}
 
+	const char *name = output->wlr_output->name;
+	char id[128];
+	output_get_identifier(id, sizeof(id), output);
+
+	size_t size = snprintf(NULL, 0, "%s on %s", id, name) + 1;
+	char *id_on_name = malloc(size);
```

Comment:
Wouldnt it be easier to have `id_on_name` on the stack with a big size? Like `PATH_MAX` or whatever's big enough

Reply:
Why not, it avoids calling `snprintf` twice to get the size, `id_on_name` would always be big enough and clear of any garbage if initialized with `= {0}`.

Reply:
I've added helpers and converted everything to them.